includes/Checker/Preparations/Universal_Runtime_Preparation.php

```diff
@@ -0,0 +1,69 @@
+<?php
+/**
+ * Class WordPress\Plugin_Check\Checker\Preparations\Universal_Runtime_Preparation
+ *
+ * @package plugin-check
+ */
+
+namespace WordPress\Plugin_Check\Checker\Preparations;
+
+use WordPress\Plugin_Check\Checker\Check_Context;
+use WordPress\Plugin_Check\Checker\Preparation;
+use Exception;
+use function PHPUnit\Framework\callback;
+
+/**
+ * Class handle all preparations required for when at least one `Runtime_Check` is being run.
+ *
+ * @since n.e.x.t
+ */
+class Universal_Runtime_Preparation implements Preparation {
+
+	/**
+	 * Context for the plugin to check.
+	 *
+	 * @since n.e.x.t
+	 * @var Check_Context
+	 */
+	protected $check_context;
+
+	/**
+	 * Sets the context for the plugin to check.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param Check_Context $check_context Check context instance for the plugin.
+	 */
+	public function __construct( Check_Context $check_context ) {
+		$this->check_context = $check_context;
+	}
+
+	/**
+	 * Runs the preparations for the theme and plugin. And returns a cleanup function.
```

Comment:
Adding a bit of clarification here. Most importantly, we should explain that this needs to be run early.
```suggestion
	 * Runs preparation step for the environment by modifying the plugins and theme to use,
	 * and returns a closure as a cleanup function.
	 *
	 * This preparation needs to be called very early in the WordPress lifecycle, before
	 * plugins are loaded, e.g. from a drop-in like `object-cache.php`.
```